marlowe-benchmark/app/Main.hs

```diff
@@ -4,33 +4,66 @@ module Main (
   main,
 ) where
 
+import Cardano.Api (
+  AsType (AsAddressAny, AsPaymentExtendedKey, AsSigningKey),
+  File (File),
+  deserialiseAddress,
+  readFileTextEnvelope,
+ )
+import Control.Monad ((<=<))
 import Data.Aeson (eitherDecodeFileStrict')
 import Data.Default (def)
+import Data.Text (Text)
 import Data.Version (showVersion)
+import Data.Word (Word32)
 import Language.Marlowe.Runtime.Benchmark (measure)
+import Language.Marlowe.Runtime.Cardano.Api (fromCardanoAddressAny)
 import Language.Marlowe.Runtime.Client (connectToMarloweRuntime)
 import Paths_marlowe_benchmark (version)
 
+import qualified Cardano.Api as C
 import qualified Options.Applicative as O
 
 -- | Execute the benchmarks.
 main :: IO ()
 main =
   do
-    (host, port, config) <- O.execParser commandParser
+    (host, port, config, faucet, out) <- O.execParser commandParser
     config' <-
       case config of
         Nothing -> pure def
         Just config'' -> either error id <$> eitherDecodeFileStrict' config''
-    connectToMarloweRuntime host (fromIntegral port) $ measure config'
+    faucet' <-
+      case faucet of
+        Just (node, magic, address, key) ->
+          do
+            address' <-
+              maybe (error "Failed to parse faucet address.") (pure . fromCardanoAddressAny) $
+                deserialiseAddress AsAddressAny address
+            key' <-
+              either (error . show) pure
+                <=< readFileTextEnvelope (AsSigningKey AsPaymentExtendedKey)
+                $ File key
+            pure $ Just (C.File node, C.BabbageEra, C.Testnet $ C.NetworkMagic magic, address', key')
+        Nothing -> pure Nothing
+    connectToMarloweRuntime host (fromIntegral port) $ measure config' faucet' out
 
-commandParser :: O.ParserInfo (String, Int, Maybe FilePath)
+commandParser :: O.ParserInfo (String, Int, Maybe FilePath, Maybe (FilePath, Word32, Text, FilePath), Maybe FilePath)
```

Comment:
Maybe it's time to add a record here

Reply:
Indeed, it kept growing and I was lazy. Fixed in 8a80f88faa19b724a56afdc61ab8338ba9db85aa.